### PR TITLE
[JFG-1018] Invoker now can be provided through providers.

### PR DESCRIPTION
**Pull request by a-badaev**

### CODE REVIEW

File: archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java [hunk 1]
```diff
@@ -45,90 +40,98 @@
...
 public class ExampleJLoadScenarioProvider extends JaggerPropertiesProvider {
-
-    // begin: following section is used for docu generation - Detailed load test scenario configuration
+    
+    
     @Bean
     public JLoadScenario exampleJaggerLoadScenario() {
-
-        // Example of using JaggerPropertiesProvider
-        String testDefinitionComment = getTestPropertyValue("example.jagger.test.definition.comment");
```
Comment (anton-antonenko) [inline archetype-java-builders/src/main/resources/archetype-resources/src/main/java/ExampleJLoadScenarioProvider.java]: Why did you remove that?

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java [hunk 2]
```diff
@@ -100,15 +101,16 @@ public Builder withQueryProvider(Iterable queryProvider) {
         }
 
         /**
-         * Optional: Sets subtypes of {@link com.griddynamics.jagger.invoker.Invoker}
-         *
-         * Instances of this class will be used to during Jagger test execution to send requests to the System under test. @n
+         * Optional: Sets a subtype of {@link com.griddynamics.jagger.engine.e1.Provider<Invoker>}
+         * @n
+         * Instances of this class will be used to during Jagger test execution to send requests to the System under test.
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: fixed

File: chassis/coordinator/src/main/java/com/griddynamics/jagger/coordinator/DefaultNodeContext.java [hunk 3]
```diff
@@ -46,11 +46,6 @@ public NodeId getId() {
         if (services.containsKey(clazz)) {
             return (T) services.get(clazz);
         }
-        try {
-            return clazz.newInstance();
-        } catch (Exception e) {
-            log.warn("Didn't manage to create an instance of {}", clazz, e);
-        }
```
Comment (a-badaev) [inline chassis/coordinator/src/main/java/com/griddynamics/jagger/coordinator/DefaultNodeContext.java]: it is not needed any more. now providers are used to create instances not reflection

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenarioFactory.java [hunk 4]
```diff
@@ -31,11 +32,17 @@
...
+    
+    private Provider<Invoker> invokerProvider;
 
     @Override
     public Scenario<Q, R, E> get(NodeContext nodeContext) {
         Invoker<Q, R, E> invoker = nodeContext.getService(invokerClazz);
-        if(invoker == null) {
+        if (invokerProvider != null) {
+            invoker = invokerProvider.provide();
+        }
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenarioFactory.java]: context is for non Java based configuration

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/v2/DefaultInvokerProvider.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * of provided subclass of {@link Invoker} to create it's instances.
+ * @n
+ * Created by Andrey Badaev
+ * Date: 30/12/16
+ */
+public class DefaultInvokerProvider implements Provider<Invoker> {
+    
+    private final Class<? extends Invoker> invokerClass;
+    
+    public DefaultInvokerProvider(final Class<? extends Invoker> invokerClass) {this.invokerClass = invokerClass;}
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/v2/DefaultInvokerProvider.java]: no

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenarioFactory.java (hunk 4 above)
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenarioFactory.java]: please add a comment here to remove request to context after JFG-1090

Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/QueryPoolScenarioFactory.java]: done
